updated structure, variable names and descriptions

badbwoi opened the pull request:
and added optional_naval_wreckage, core_mexes, extra_mexes, jamming

badbwoi committed:
updated structure, variable names and some descriptions

'used slots' (dynamic_spawn = 2) replaces 'mirrormex'
'dupicatesinglemex' replaced by all 3 crazy rush options
Civilian Base and Civilian Defenses separated
renamed 'optional_reclaim' to 'optional_wreckage'
added optional_naval_wreckage, core_mexes, extra_mexes, jamming

badbwoi committed:
updated structure, variable names and some descriptions

changed variableNames to improve workflow - was constantly confused by name changes (mex or mass) for same option
added info how to use tables

badbwoi committed:
updated structure, variable names and some LOGs

constant naming for linked variables in script, options and tables
options.lua: feature_name
table.lua: featureName

renamed automex to dynamic_spawn
'used slots' replaces mirrormex option and is now bound to dynamic_spawn = 2

removed crazyrush from dynamic_spawn
renamed dupicatesinglemex to forwardCrazyrushMexes
crazyrush_mexes enables now 1 of 3 crazyrush options

renamed optional_reclaim to optional_wreckage
added optional_naval_wreckage, optional_civilian_defenses, core_mexes, extra_mexes, jamming

renamed tree to TreeRegrowSpeed

badbwoi committed:
renamed additional_hydro (former hydroplus) to additional_hydros

badbwoi committed:
renamed additional_hydro (former hydroplus) to additional_hydros

CookieNoob commented on mapname_options.lua:
```diff
@@ -2,77 +2,61 @@ options ={
-        { text = "enabled - crazyrush 1 mex", help = "Map will activate the crazyrush script and spawn 1 starting mex for all players.", key = 6, },
-        { text = "enabled - crazyrush", help = "Map will activate the crazyrush script.", key = 7, },
+        { text = "mirror slots", help = "Spawn resources for player & mirror slot.", key = 1, },
+        { text = "used slots", help = "Spawn resources for player on used slots.", key = 2, },
```
diese option ist problematisch, da damit die map unbalanced wird...

CookieNoob commented on mapname_options.lua:
```diff
@@ -2,77 +2,61 @@ options ={
+        { text = "2v2 setup", help = "Don't adjust for player & spawn resources for 2v2.", key = 3, },
+        { text = "4v4 setup", help = "Don't adjust for player & spawn resources for 4v4.", key = 4, },
+        { text = "6v6 setup", help = "Don't adjust for player & spawn resources for 6v6.", key = 5, },
+        { text = "8v8 setup", help = "Don't adjust for player & spawn resources for maximum player count.", key = 6, },
```
diese optionen würde ich zumindestens teilweise auskommentieren

CookieNoob commented on mapname_options.lua:
```diff
@@ -2,77 +2,61 @@ options ={
-        { text = "enabled", help = "Only spawn mex/hydro when opposing spot is also taken. This option is not recommended, however it can make uneven matches fairer.", key = 2, },
-
-    },
-},
```
wieso hast du diese option entfernt?

CookieNoob commented on mapname_options.lua:
```diff
@@ -81,22 +65,24 @@ options ={
-    help = "makes a few mexes behave like crazyrush mexes.",
-    key = 'dupicatesinglemex',
-    pref = 'dupicatesinglemex',
+    label = "Crazyrush",
```
diese option ist "immer funktionsfähig", daher sollte sie nicht im auskommentierten bereich stehen

CookieNoob replied:
die option forward sollte auskommentiert sein

CookieNoob commented on mapname_options.lua:
```diff
@@ -209,45 +195,114 @@ options ={
 
 {
-    default = 1,
+    default = 2,
```
default sollte identisch mit dem default im script sein (da steht zwar 2, aber das wird zu einem crash führen wenn keine civs definiert sind auch wenn die option auskommentiert wird)
ändern zu 1

CookieNoob commented on mapname_script.lua:
```diff
@@ -1,27 +1,35 @@
+local underwater_mexes = ScenarioInfo.Options.underwater_mexes or 1
+local back_mexes = ScenarioInfo.Options.back_mexes or 1
+local island_mexes = ScenarioInfo.Options.island_mexes or 1
+local optional_wreckage = ScenarioInfo.Options.optional_wreckage or 2
```
default ändern auf 1

CookieNoob commented on mapname_script.lua:
```diff
@@ -1,27 +1,35 @@
+local back_mexes = ScenarioInfo.Options.back_mexes or 1
+local island_mexes = ScenarioInfo.Options.island_mexes or 1
+local optional_wreckage = ScenarioInfo.Options.optional_wreckage or 2
+local optional_naval_wreckage = ScenarioInfo.Options.optional_naval_wreckage or 2
```
default ändern auf 1

CookieNoob commented on mapname_script.lua:
```diff
@@ -1,27 +1,35 @@
+local island_mexes = ScenarioInfo.Options.island_mexes or 1
+local optional_wreckage = ScenarioInfo.Options.optional_wreckage or 2
+local optional_naval_wreckage = ScenarioInfo.Options.optional_naval_wreckage or 2
+local optional_civilian_base = ScenarioInfo.Options.optional_civilian_base or 2
```
default ändern auf 1

CookieNoob commented on mapname_script.lua:
```diff
@@ -1,27 +1,35 @@
+local optional_wreckage = ScenarioInfo.Options.optional_wreckage or 2
+local optional_naval_wreckage = ScenarioInfo.Options.optional_naval_wreckage or 2
+local optional_civilian_base = ScenarioInfo.Options.optional_civilian_base or 2
+local optional_civilian_defenses = ScenarioInfo.Options.optional_civilian_defenses or 2
```
default ändern auf 1

CookieNoob commented on mapname_script.lua:
```diff
@@ -1,27 +1,35 @@
+local crazyrush_mexes = ScenarioInfo.Options.crazyrush_mexes or 1
+local additional_mexes = ScenarioInfo.Options.additional_mexes or 1
+local core_mexes = ScenarioInfo.Options.core_mexes or 1
+local extra_mexes = ScenarioInfo.Options.extra_mexes or 2
```
default ändern auf 1

CookieNoob commented on mapname_script.lua:
```diff
@@ -30,42 +38,151 @@ local MexList ={}
+end
+
+--OnStart
+function OnStart()
```
OnStart sollte der Übersichtlichkeit halber über den startCrazy... und startGrowing... funktionen stehen

CookieNoob commented on mapname_script.lua:
```diff
@@ -30,42 +38,151 @@ local MexList ={}
+    --check if a message needs to be displayed
+    ForkThread(showmessage)
+    
+--[[    --remove some of the rocks
```
dieser kommentarbereich kann entfernt werden (die funktion funktioniert nicht)

CookieNoob commented on mapname_script.lua:
```diff
@@ -30,42 +38,151 @@ local MexList ={}
+--Startmessage
+function gatherFeedback()
+    WaitSeconds(10)
+    BroadcastMSG('If you see any bugs with the map, plz tell CookieNoob. Thx.',  -- message
```
'If you see any bugs with the map, plz tell CookieNoob. Thx.' ->
'If you see any bugs with the map, plz tell the map author. Thx.'

CookieNoob commented on mapname_script.lua:
```diff
@@ -30,42 +38,151 @@ local MexList ={}
+--call from OnPopulate to prevent despawning map-decal glitch (map-decal disappeared, when optional structures (probably due to decal) and a certain amount of unit footprint/track-decals were spawned)
+function OptionalUnits()
+    LOG("ADAPTIVE: Optional Units:")
+    -- LOG("ADAPTIVE: optional_reclaim_middle = ", optional_reclaim_middle)
```
diese logs sollten nicht auskommentiert sein

CookieNoob commented on mapname_script.lua:
```diff
@@ -30,42 +38,151 @@ local MexList ={}
+    
     if(optional_civilian_base > 1) then
         local spawncivs = true
         for m = 13, 14 do
```
das ist nur ein sehr spezieller fall, das sollte auskommentiert als block im code stehen, aber der normale fall sollte der code hier drunter sein (den du auskommentiert hast)

CookieNoob commented on mapname_script.lua:
```diff
@@ -75,180 +192,223 @@ function OnPopulate()
+    elseif(optional_civilian_defenses == 4) then
+        ScenarioUtils.CreateArmyGroup('ARMY_17', 'Optional_Civilian_Defenses_2', true)
+        ScenarioUtils.CreateArmyGroup('ARMY_17', 'Optional_Civilian_Defenses_3', true)
+        ScenarioUtils.CreateArmyGroup('ARMY_17', 'Optional_Civilian_Defenses_4', true)
```
```
+    if(optional_civilian_defenses == 2) then
+        ScenarioUtils.CreateArmyGroup('ARMY_17', 'Optional_Civilian_Defenses_2', true)
+    elseif(optional_civilian_defenses == 3) then
+        ScenarioUtils.CreateArmyGroup('ARMY_17', 'Optional_Civilian_Defenses_2', true)
+        ScenarioUtils.CreateArmyGroup('ARMY_17', 'Optional_Civilian_Defenses_3', true)
+    elseif(optional_civilian_defenses == 4) then
+        ScenarioUtils.CreateArmyGroup('ARMY_17', 'Optional_Civilian_Defenses_2', true)
+        ScenarioUtils.CreateArmyGroup('ARMY_17', 'Optional_Civilian_Defenses_3', true)
+        ScenarioUtils.CreateArmyGroup('ARMY_17', 'Optional_Civilian_Defenses_4', true)
```


dieser block kann einfach ersetzt werden durch
```
for i = 2, optional_civilian_defenses do
    ScenarioUtils.CreateArmyGroup('ARMY_17', 'Optional_Civilian_Defenses_' .. i, true)
```

badbwoi replied:
Hier handelt es sich um Wrecks und lebende Einheiten, die aus den Gruppen 2, 3 & 4 bestehen. Dein vorgeschlagener Style macht Probleme wenn ich die Optionen 5-7 auswähle, da diese Gruppen nicht existieren.

CookieNoob replied:
oh hab gar nicht gesehen dass das eine wracks sind xD

CookieNoob commented on mapname_script.lua:
```diff
@@ -75,180 +192,223 @@ function OnPopulate()
+    elseif(optional_civilian_defenses == 7) then
+        ScenarioUtils.CreateArmyGroup('ARMY_17', 'Optional_Civilian_Defenses_2', false)
+        ScenarioUtils.CreateArmyGroup('ARMY_17', 'Optional_Civilian_Defenses_3', false)
+        ScenarioUtils.CreateArmyGroup('ARMY_17', 'Optional_Civilian_Defenses_4', false)
```
```
+    elseif(optional_civilian_defenses == 5) then
+        ScenarioUtils.CreateArmyGroup('ARMY_17', 'Optional_Civilian_Defenses_2', false)
+    elseif(optional_civilian_defenses == 6) then
+        ScenarioUtils.CreateArmyGroup('ARMY_17', 'Optional_Civilian_Defenses_2', false)
+        ScenarioUtils.CreateArmyGroup('ARMY_17', 'Optional_Civilian_Defenses_3', false)
+    elseif(optional_civilian_defenses == 7) then
+        ScenarioUtils.CreateArmyGroup('ARMY_17', 'Optional_Civilian_Defenses_2', false)
+        ScenarioUtils.CreateArmyGroup('ARMY_17', 'Optional_Civilian_Defenses_3', false)
+        ScenarioUtils.CreateArmyGroup('ARMY_17', 'Optional_Civilian_Defenses_4', false)
```

ist eine kopie vom code darüber und macht das selbe -> entfernen

badbwoi replied:
Wie ebend beschrieben, werden hier die Einheiten lebend gespawned

CookieNoob commented on mapname_script.lua:
```diff
@@ -75,180 +192,223 @@ function OnPopulate()
-    -- get map markers
+    LOG("ADAPTIVE: Create Resources:")
+    LOG("ADAPTIVE: dynamic_spawn = ", dynamic_spawn)
+    -- LOG("ADAPTIVE: crazyrush_mexes = ", crazyrush_mexes)
```
diese logs sollten nicht auskommentiert sein

CookieNoob commented on mapname_script.lua:
```diff
@@ -75,180 +192,223 @@ function OnPopulate()
+    --FIX SETUP FOR X PLAYER
+    --2v2 setup armys 1-4 (remove 5-16)
+    elseif(dynamic_spawn == 3) then
+        for m = 5, 16 do
```
benutzt nicht "maxPlayerOnMap"

CookieNoob commented on mapname_script.lua:
```diff
@@ -75,180 +192,223 @@ function OnPopulate()
+        
+    --4v4 setup, armys 1-8 (remove 9-16)
+    elseif(dynamic_spawn == 4) then
         for m = 9, 16 do
```
benutzt nicht "maxPlayerOnMap"

CookieNoob commented on mapname_script.lua:
```diff
@@ -75,180 +192,223 @@ function OnPopulate()
+        
+    --6v6 setup, army 1-12 (remove 13-16)
+    elseif(dynamic_spawn == 5) then
+        for m = 13, 16 do
```
benutzt nicht "maxPlayerOnMap"

CookieNoob commented on mapname_script.lua:
```diff
@@ -75,180 +192,223 @@ function OnPopulate()
 
-            if(hydroplus == 1) then
-                doit=FalseIfInList(name, spwnAdditionalHydro, HydroString, doit);
+            if(additional_hydros == 1) then
```
gute idee mit dieser konvention

CookieNoob commented on mapname_script.lua:
```diff
@@ -300,45 +462,42 @@ function DoitIfInList(name, list, string_function, _doit)
+        lod,                     # LOD
+        0,                       # Duration (0 == does not expire)
+        -1,                      # army (-1 == not owned by any single army)
+        0                        # ???
```
inzwischen ist klar, dass dieser parameter "fidelity" regelt... 
-> "???"  ersetzen durch "fidelity"

CookieNoob commented on mapname_tables.lua:
```diff
@@ -45,24 +45,42 @@ spwnHydroArmy ={    {},
+extraMexes = {{}}
+
+
+
```
ich habe es jetzt nicht überprüft, aber sind alle tabellen die im code verwendet werden wirklich auch hier aufgeführt?

CookieNoob reviewed (changes requested):
die konvention mit dem unterstrich finde ich gut, aber einige kleinigkeiten müssen noch geändert werden (die default werte z.B.), da das script sonst abstürzt (auf einer neu erstellten map, also bevor änderungen gemacht wurden wie z.B. civ gruppen hinzugefügt werden)

badbwoi committed:
applied changes

moved OnStart, my starttext, added maxPlayerOnMap to all dynamic_spawn, changed civilian base and defenses

restructure (sorted variables to match table)

renamed additional_mexes/additionalMexes to base_mexes/baseMexes

removed rockRemove

renamed optional_reclaim_middle to optional_wreckage_middle

improved AddFactionReclaimBack to use 1 instead of 2 variables: renamed optional_reclaim_back and optional_reclaim_adaptive to optional_adaptive_faction_wreckage

back_mexes/backMexes to expansion_mexes/expansionMexes

all defaults to 1 (except expand_map and naturalReclaimModifier in options)

added if _mexname is string

added 'end' to end OptionalUnits function

badbwoi committed:
applied changes, added sorting and comments

badbwoi committed:
applied changes

badbwoi committed:
added missing [e] to remove baseMexes

badbwoi committed:
'no mirror = no resources' to dynamic_spawn

badbwoi committed:
'no mirror = no resources' to dynamic_spawn

badbwoi committed:
no tabs

CookieNoob commented on mapname_script.lua:
```diff
@@ -503,8 +517,8 @@ end
-    local mexsidetop = ScenarioInfo.Options.mexsidetop or 1
-    local mexsidebot = ScenarioInfo.Options.mexsidebot or 1
+    local mexsidetop = ScenarioInfo.Options.mexsidetop or 1 --not in options?
+    local mexsidebot = ScenarioInfo.Options.mexsidebot or 1 --not in options?
```
die map "Adaptive Wonder Open" hat oben rechts und unten links jeweils einzeln anpassbare mexe, daher kommt das hier... vielleicht einen kommentar dazu und sonst einfach drin lassen :)

CookieNoob commented:
sieht gut aus
